tests/unit/test_link_generation_facade.py

```diff
@@ -0,0 +1,130 @@
+    @pytest.mark.jira('asc-239')
+    def test_verify_network_list(host):
+        \"\"\"Verify the neutron network was created\"\"\"
+        cmd = pre_cmd + "openstack network list\""
+        output = host.run(cmd)
+&gt;       assert ("GATEWAY_NET" in output.stdout)
+E       assert 'GATEWAY_NET' in ''
+E        +  where '' = CommandResult(command='bash -c "source /root/openrc; openstack network list"', exit_status=0, stdout=u'', stderr=u'').stdout
+
+tests/test_verify_networks_created.py:23: AssertionError
+""".format(LINE_NUMBER)  # noqa
+
+TEST_FILE = 'tests/totally_real/verify_I_am_good_link.py'
+
+
+# ======================================================================================================================
+# Test Suites
+# ======================================================================================================================
+class TestLinkGenerationFacade(object):
+    """Tests for the LinkGenerationFacade"""
+
+    def test_asc_failure_link(self, mocker):
+        """Validate when configured with asc as ci-environment"""
+        zz = mocker.MagicMock()
+        zz.ci_environment = 'asc'
+        zz.testsuite_props = ASC_TESTSUITE_PROPS
+
+        zztl = mocker.MagicMock()
+        zztl.test_file = TEST_FILE
+        zztl.failure_output = FAILURE_OUTPUT
+
+        lgf = LinkGenerationFacade(zz)
+        failure_link = lgf.github_testlog_failure_link(zztl)
+        assert failure_link == 'https://github.com/rcbops/molecule-validate-neutron-deploy/blob/pike-rc/molecule/default/tests/totally_real/verify_I_am_good_link.py#L23'  # noqa
+
+    def test_asc_def_link(self, mocker):
+        """Validate fallback to def line number"""
+        zz = mocker.MagicMock()
+        zz.ci_environment = 'asc'
+        zz.testsuite_props = ASC_TESTSUITE_PROPS
+
+        zztl = mocker.MagicMock()
+        zztl.test_file = TEST_FILE
+        zztl.failure_output = 'I dont contain an assert message'
+        zztl.def_line_number = '88'
+
+        lgf = LinkGenerationFacade(zz)
+        failure_link = lgf.github_testlog_failure_link(zztl)
+        assert failure_link == 'https://github.com/rcbops/molecule-validate-neutron-deploy/blob/pike-rc/molecule/default/tests/totally_real/verify_I_am_good_link.py#L88'  # noqa
+
+    def test_asc_file_link(self, mocker):
+        """Validate fallback to file with no line number"""
+        zz = mocker.MagicMock()
+        zz.ci_environment = 'asc'
+        zz.testsuite_props = ASC_TESTSUITE_PROPS
+
+        zztl = mocker.MagicMock()
+        zztl.test_file = TEST_FILE
+        zztl.failure_output = 'I dont contain an assert message'
+        zztl.def_line_number = ''
+
+        lgf = LinkGenerationFacade(zz)
+        failure_link = lgf.github_testlog_failure_link(zztl)
+        assert failure_link == 'https://github.com/rcbops/molecule-validate-neutron-deploy/blob/pike-rc/molecule/default/tests/totally_real/verify_I_am_good_link.py'  # noqa
```

Comment:
See comment for line 58.

FWIW, you could also break this line to fit within the flake8 line limit value using an expression to compose the URI.
```
assert failure_link == ('https://github.com/'
                        'rcbops/'
                        'molecule-validate-neutron-deploy/'
                        'blob/pike-rc/'
                        'molecule/default/'
                        'tests/totally_real/verify_I_am_good_link.py')
```

Reply:
Fixed